ui/src/track-layout/track-layout-model.tsx

```diff
@@ -179,6 +180,43 @@ export function getNameSpecifier(
     return nameStructure && 'specifier' in nameStructure ? nameStructure.specifier : undefined;
 }
 
+export const locationTrackNameFieldsSanitized = (
+    saveRequest: LocationTrackSaveRequest,
+): Pick<LocationTrackSaveRequest, 'nameFreeText' | 'nameSpecifier'> => {
+    const freeText = !isNilOrBlank(saveRequest.nameFreeText)
+        ? saveRequest.nameFreeText?.trim()
+        : undefined;
+    const specifier = saveRequest.nameSpecifier;
+
+    switch (saveRequest.namingScheme) {
+        case LocationTrackNamingScheme.FREE_TEXT:
+        case LocationTrackNamingScheme.WITHIN_OPERATING_POINT:
+            return {
+                nameFreeText: freeText,
+                nameSpecifier: undefined,
+            };
+        case LocationTrackNamingScheme.TRACK_NUMBER_TRACK:
+            return {
+                nameFreeText: freeText,
+                nameSpecifier: specifier,
+            };
+        case LocationTrackNamingScheme.BETWEEN_OPERATING_POINTS:
+            return {
+                nameFreeText: undefined,
+                nameSpecifier: specifier,
+            };
+        case LocationTrackNamingScheme.CHORD:
+            return {
+                nameFreeText: undefined,
+                nameSpecifier: undefined,
+            };
+        case undefined:
+            throw Error('Naming scheme is mandatory for location track save request!');
+        default:
+            return exhaustiveMatchingGuard(saveRequest.namingScheme);
+    }
+};
```

Comment:
GVT-3210: Aiemmin siis bäkkärille lähti kaikki nimeen liittyvät kentät riippumatta kuuluvatko ne kyseisen nimeämisskeeman tietoihin vai eivät. Täällä siis käytännössä katsotaan mitä saa ja ei saa olla, ja palautetaan sen perusteella `LocationTrackSaveRequest`:in fragmentti (sillä idealla että se vaan mergetään aiemman päälle). Mietin myös mergen tekemistä täällä ja kokonaan uuden `SaveRequest`:in palauttamista, mutta se johti koodillisiin ärsyttävyyksiin, joten valitsin tällaisen tien. Not really happy, olisipa kotlinin palauttava `when`